src/components/attribute-picker/attribute-picker.js

```diff
@@ -426,10 +426,18 @@ class AttributePicker extends ArrowKeysMixin(LocalizeLabsElement(LitElement)) {
 	}
 
 	_onInputTextChanged(e) {
+		e.stopPropagation();
+
 		this._text = e.target.value;
 		if (this._dropdownIndex >= 0) {
 			this.allowFreeform ? this._dropdownIndex = -1 : this._dropdownIndex = 0;
 		}
+
+		this.dispatchEvent(new CustomEvent('d2l-labs-attribute-picker-text-changed', {
+			detail: { text: this._text },
+			bubbles: true,
+			composed: true,
```

Comment:
In general, I typically recommend _not_ bubbling up through the DOM or being composed unless you absolutely need it. It's just nicer to not having events flying around past nodes that will never need to care about them. That would still allow a consumer to wire up a listener directly to the attribute picker, which is likely your use case here.

Reply:
Gotcha, that makes sense. I had enabled `bubbles` and `composed` for consistency with the other two event types in this component, but it's true that our use-case has no need for the event to bubble up further than 1 level. I've tested this change just now and it works fine, so I'll push a commit shortly.